### DISCUSSION TITLE
Support for Pydantic model serialization in `astream_log` calls

### DISCUSSION THREAD

**Discussion by edochi** [langchain-ai/langserve#616] [Ideas]
Making some tests I noticed that Pydantic objects included in dictionaries passing through a chain produce an error when the `astream_log` is used on a RemoteRunnable.

```
TypeError: Type is not JSON serializable: <ObjectName>
```

An easy workaround is to use `model_dump` in the chain to transform it in to a dictionary, and move it along the chain, or to change it in a dataclass or any other data structure that is supported by `orjson.dump()`. 

However, in my opinion, the best solution would be to add a direct support for Pydantic model serialization, given their widespread usage, even more in this kind of applications, where the validation of payloads / data is crucial.

I think this issue could be considered related to #413, but I decided to open a discussion instead since I don't think it's an issue by itself

**Comment by eyurtsev** [comment]
It'll serialize correctly, but the de-serialization will not work automatically?

If we expose to the user control over the serializer, the user will be able to take care of de-serialization as well
